lib/Doctrine/DBAL/Driver/PDOSqlsrv/Driver.php

```diff
@@ -61,16 +66,58 @@ private function _constructPdoDsn(array $params)
         }
 
         if (isset($params['dbname'])) {
-            $dsn .= ';Database=' .  $params['dbname'];
+            $dsn .= ';Database=' . $params['dbname'];
         }
 
-        if (isset($params['MultipleActiveResultSets'])) {
-            $dsn .= '; MultipleActiveResultSets=' . ($params['MultipleActiveResultSets'] ? 'true' : 'false');
+        if (! empty($connectionOptions)) {
+            $dsn .= $this->getConnectionOptionsDsn($connectionOptions);
         }
 
         return $dsn;
     }
 
+    /**
+     * Separates a connection options from a driver options
+     *
+     * @param array $options
+     * @return array
+     */
+    private function splitOptions(array $options)
+    {
+        $driverOptions     = [];
+        $connectionOptions = [];
+
+        foreach ($options as $optionKey => $optionValue) {
+            if (is_int($optionKey)) {
+                $driverOptions[$optionKey] = $optionValue;
+            } else {
+                $connectionOptions[$optionKey] = $optionValue;
+            }
+        }
+
+        return [
+            'driverOptions' => $driverOptions,
```

Comment:
Given it's a private API, instead of introducing named keys, would it be cleaner to just use a tuple? Like:
```php
function splitOptions()
{
    return [$connectionOptions, $driverOptions];
}

[$connectionOptions, $driverOptions] = splitOptions();
```